Northstar.Client/mod/scripts/vscripts/ui/menu_mod_settings.nut

```diff
@@ -0,0 +1,1066 @@
+}
+
+void function ResetConVar( var button )
+{
+	print("ok")
+	ConVarData conVar = file.filteredList[ int ( Hud_GetScriptID( Hud_GetParent( button ) ) ) + file.scrollOffset ]
+
+	if ( conVar.isCategoryName )
+		ShowAreYouSureDialog( "#ARE_YOU_SURE", ResetAllConVarsForModEventHandler( conVar.catName ), "#WILL_RESET_ALL_SETTINGS"  )
+	else ShowAreYouSureDialog( "#ARE_YOU_SURE", ResetConVarEventHandler( int ( Hud_GetScriptID( Hud_GetParent( button ) ) ) + file.scrollOffset ), Localize( "#WILL_RESET_SETTING", Localize( conVar.displayName ) ) )
+}
+
+void function ShowAreYouSureDialog( string header, void functionref() func, string details )
+{
+	DialogData dialogData
+	dialogData.header = header
+	dialogData.message = details
+	
+	AddDialogButton( dialogData, "#NO" )
+	AddDialogButton( dialogData, "#YES", func )
+
+	AddDialogFooter( dialogData, "#A_BUTTON_SELECT" )
+	AddDialogFooter( dialogData, "#B_BUTTON_BACK" )
+
+	OpenDialog( dialogData )
+}
+
+void functionref() function ResetAllConVarsForModEventHandler( string catName )
+{
+	return void function() : ( catName )
+	{
+		for ( int i = 0; i < file.conVarList.len(); i++ )
+		{
+			ConVarData c = file.conVarList[i]
+			if ( c.catName != catName || c.isCategoryName || c.isEmptySpace )
+				continue
+			SetConVarToDefault( c.conVar )
+
+			int index = file.filteredList.find( c )
+			if ( file.filteredList.find( c ) < 0 )
+				continue
+
+			if ( min( BUTTONS_PER_PAGE, max( 0, index - file.scrollOffset ) ) == index - file.scrollOffset )
+				Hud_SetText( Hud_GetChild( file.modPanels[ i - file.scrollOffset ], "TextEntrySetting" ), c.isEnumSetting ? c.values[ GetConVarInt( c.conVar ) ] : GetConVarString( c.conVar ) )
+		}
+	}
+}
+
+void functionref() function ResetConVarEventHandler( int modIndex )
+{
+	return void function() : ( modIndex )
+	{
+		ConVarData c = file.filteredList[ modIndex ]
+		SetConVarToDefault( c.conVar )
+		if ( min( BUTTONS_PER_PAGE, max( 0, modIndex - file.scrollOffset ) ) == modIndex - file.scrollOffset )
+			Hud_SetText( Hud_GetChild( file.modPanels[ modIndex - file.scrollOffset ], "TextEntrySetting" ), c.isEnumSetting ? c.values[ GetConVarInt( c.conVar ) ] : GetConVarString( c.conVar ) )
+	}
+}
+
+////////////
+// slider //
+////////////
+void function UpdateMouseDeltaBuffer( int x, int y )
+{
+	mouseDeltaBuffer.deltaX += x
+	mouseDeltaBuffer.deltaY += y
+
+	SliderBarUpdate()
+}
+
+void function FlushMouseDeltaBuffer()
+{
+	mouseDeltaBuffer.deltaX = 0
+	mouseDeltaBuffer.deltaY = 0
+}
+
+void function SliderBarUpdate()
+{
+	if ( file.filteredList.len() <= 15 )
+	{
+		FlushMouseDeltaBuffer()
+		return
+	}
+
+	var sliderButton = Hud_GetChild( file.menu, "BtnModListSlider" )
+	var sliderPanel = Hud_GetChild( file.menu, "BtnModListSliderPanel" )
+	var movementCapture = Hud_GetChild( file.menu, "MouseMovementCapture" )
+
+	Hud_SetFocused( sliderButton )
+
+	float minYPos = -40.0 * ( GetScreenSize()[1] / 1080.0 ) // why the hardcoded positions?!?!?!?!?!
```

Comment:
sounds like a giga skill issue and should be possible without hard coding
(In a future PR please I want this merged and it's not breaking anything)

Reply:
The `1080` sounds like using it on a different resolution than `1080p` might break things. Did anyone test on higher/lower resolutions?

Reply:
> The 1080 sounds like using it on a different resolution than 1080p might break things.

This code looks like it aims to avoid exactly that, `GetScreenSize()[1]` would also be `1080` in `1080p`, this is finding the size multiplier needed for the variable on different resolutions, as it is built and designed in `1080p` in the `.menu` and .res files, and then scaled accordingly where needed (things like rui dont obey stuff like %20 properly in `.menu` and `.res` files, they do it based on screen size, not parent element size, so things get bad rather fast)

Reply:
To be fair, Respawn themselves only calculate with 1920x1080px so this won't be an issue

Reply:
resolving - doesn't matter anyways. might add in a future PR.